certbot/storage.py

```diff
@@ -198,6 +209,70 @@ def lineagename_for_filename(config_filename):
             "renewal config file name must end in .conf")
     return os.path.basename(config_filename[:-len(".conf")])
 
+def renewal_filename_for_lineagename(config, lineagename):
+    """Returns the lineagename for a configuration filename.
+    """
+    return os.path.join(config.renewal_configs_dir, lineagename) + ".conf"
+
+def _full_archive_path(config_obj, cli_config, lineagename):
+    """Returns the full archive path for a lineagename"""
+    if "archive_dir" in config_obj:
+        return config_obj["archive_dir"]
+    else:
+        return os.path.join(cli_config.default_archive_dir, lineagename)
+
+def delete_files(config, certname):
+    """Delete all files related to the certificate.
+
+    If some files are not found, ignore them and continue.
+    """
+    renewal_filename = renewal_file_for_certname(config, certname)
+    try:
+        renewal_config = configobj.ConfigObj(renewal_filename)
+    except configobj.ConfigObjError:
+        raise errors.CertStorageError(
```

Comment:
That might be the right thing to do 99% of the time, but it will severely mess up people's configs the other 1%.

To rename another lineage to the same name, they would have had to do it manually rather than through certbot rename. Which means they're already messaging around with files manually, which makes it more likely that this will mess things up.

I'd be good with printing a message saying they may want to delete the default files here.